src/lib/functions/runtimes/js/index.mjs

```diff
@@ -48,12 +51,49 @@ export const getBuildFunction = async ({ config, directory, errorExit, func, pro
   // main file otherwise.
   const functionDirectory = dirname(func.mainFile)
   const srcFiles = functionDirectory === directory ? [func.mainFile] : [functionDirectory]
-  const schedule = await parseForSchedule({ mainFile: func.mainFile, config, projectRoot })
 
-  return () => ({ schedule, srcFiles })
+  return () => ({ schedule: metadata.schedule, srcFiles })
 }
 
+const workerURL = new URL('worker.mjs', import.meta.url)
+
 export const invokeFunction = async ({ context, event, func, timeout }) => {
+  const workerData = {
+    clientContext: JSON.stringify(context),
+    event,
+    // If a function builder has defined a `buildPath` property, we use it.
+    // Otherwise, we'll invoke the function's main file.
+    // Because we use import() we have to use file:// URLs for windows
+    lambdaPath: pathToFileURL((func.buildData && func.buildData.buildPath) || func.mainFile).href,
+    timeout,
+  }
+
+  const worker = new Worker(workerURL, { workerData })
+  return await new Promise((resolve, reject) => {
+    worker.on('message', (result) => {
+      if (result?.streamPort) {
+        const client = createConnection(
+          {
+            port: result.streamPort,
+            host: 'localhost',
+          },
+          () => {
+            result.body = client
+            resolve(result)
+          },
+        )
+        client.on('error', reject)
+      } else {
+        resolve(result)
+      }
+    })
+
+    worker.on('error', reject)
+  })
+}
+
+// unused right now
```

Comment:
My understanding is that we'll hook this up before merging, so that the new method is only used for functions using the v2 API? Or at the very least use a feature flag. I think we should try to avoid moving every single invocation to the new method at once.

Reply:
done in https://github.com/netlify/cli/pull/5804/commits/71cb338c5347b86d9f97bea49ed45b128b9171e6